test/jdk/java/awt/event/ComponentEvent/ComponentEventTest.java

```diff
@@ -0,0 +1,325 @@
+
+    private static void initializeGUI() {
+        frame = new Frame("Component Event Test");
+        frame.setLayout(new FlowLayout());
+
+        Panel panel = new Panel();
+        Button button = new Button("Button");
+        Label label = new Label("Label");
+        List list = new List();
+        list.add("One");
+        list.add("Two");
+        list.add("Three");
+        Choice choice = new Choice();
+        choice.add("Red");
+        choice.add("Orange");
+        choice.add("Yellow");
+        Checkbox checkbox = new Checkbox("Checkbox");
+        Scrollbar scrollbar = new Scrollbar(Scrollbar.HORIZONTAL, 0, 1, 0, 255);
+        TextField textfield = new TextField(15);
+        TextArea textarea = new TextArea(5, 15);
+
+        components = new Component[] { panel, button, label, list, choice,
+            checkbox, scrollbar, textfield, textarea, frame };
+
+        for (int i = 0; i < components.length - 1; i++) {
+            components[i].addComponentListener(componentListener);
+            frame.add(components[i]);
+        }
+        frame.addComponentListener(componentListener);
+
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    public static void main(String[] args) throws Exception {
+        try {
+            robot = new Robot();
+            robot.setAutoWaitForIdle(true);
+
+            EventQueue.invokeAndWait(ComponentEventTest::initializeGUI);
+            robot.waitForIdle();
+            robot.delay(DELAY);
+
+            doTest();
+
+            System.out.println("Test PASSED");
+        } finally {
+            EventQueue.invokeAndWait(ComponentEventTest::disposeFrame);
+        }
+    }
+
+    private static void doTest()
+        throws InvocationTargetException, InterruptedException {
+
+        EventQueue.invokeAndWait(() -> {
+            compAt = components[0].getLocationOnScreen();
+            compSize = components[0].getSize();
+        });
+
+        robot.mouseMove(compAt.x + compSize.width / 2,
+            compAt.y + compSize.height / 2);
+
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+        for (int i = 0; i < components.length; i++) {
+            for (boolean state : new boolean[] { true, false }) {
+                doTest(i, state);
+            }
+        }
+
+        robot.delay(DELAY);
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            frame.setExtendedState(Frame.ICONIFIED);
+        });
+
+        robot.delay(DELAY);
+        if (componentShown || componentHidden || componentMoved
+            || componentResized) {
+            printErrEvents("ComponentEvent triggered when frame is iconified");
+        }
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            frame.setExtendedState(Frame.NORMAL);
+        });
+
+        robot.delay(DELAY);
+        if (componentShown || componentHidden) {
```

Comment:
Is it possible that any other event is triggered? Or do you expect other events?

When you inconify the frame, you verify of all the event flags but you verify only two flags here.

Reply:
When inconify i expect all events should not triggered , but when form Frame.ICONIFIED to Frame.NORMAL componentMoved and componentResized events are triggered. Events order on Windows and Mac are different.

Order of the events for Frame from Frame.ICONIFIED to Frame.NORMAL

On Windows

Component Shown 
Frame is Visible with ExtendedState as ICONIFIED
Component Moved
Component Resized
Frame is Visible with ExtendedState as NORMAL

On MAC
Component Moved
Component Resized
Component Shown 
Frame is Visible with ExtendedState as ICONIFIED
Frame is Visible with ExtendedState as NORMAL

Reply:
We observed different behavior between MS Windows and other OS ,we natively received WM_SIZE/WM_MOVE events when set frame iconify to
deiconify , The AWT sends the events to components when it receives the events from the native system. 

Now we are applying the OS specific check to test this scenario.

Please check the details at  JDK-6754618